sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -8992,13 +9005,19 @@ __ESIMD_API
   constexpr lsc_data_order Transposed = lsc_data_order::nontranspose;
   using MsgT = typename lsc_expand_type<T>::type;
   constexpr int IOp = lsc_to_internal_atomic_op<T, Op>();
-  simd<MsgT, N> Src0Msg = lsc_format_input<MsgT>(src0);
   auto si = get_surface_index(acc);
-  simd<MsgT, N> Tmp =
-      __esimd_lsc_xatomic_bti_1<MsgT, IOp, L1H, L2H, AddressScale, ImmOffset,
-                                EDS, VS, Transposed, N>(
-          pred.data(), byte_offset.data(), Src0Msg.data(), si);
-  return lsc_format_ret<T>(Tmp);
+  if constexpr (std::is_same_v<T, double> || std::is_same_v<T, float>) {
+    return __esimd_lsc_xatomic_bti_1<T, IOp, L1H, L2H, AddressScale, ImmOffset,
```

Comment:
@vmustya - The changes here look correct, just a question to VC/GenX, are `bti` intrinsics also emulated fp FP and need true FP types instead of integer types, right?

Reply:
We don't emulate any BTI atomics.

Reply:
Got it, but changes here shouldn't break anything and at least will be consistent with the USM api.